### PR TITLE
Fix edge case when haplotypes have leading insertion after trimming

### PR DESCRIPTION
**Pull request by davidbenjamin**
@takutosato Two quick edge cases and new unit tests.  These were exposed when fixing other bugs.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/utils/read/AlignmentUtils.java [hunk 1]
```diff
@@ -141,6 +141,9 @@ public static GATKRead createReadAlignedToRef(final GATKRead originalRead,
      *
      * If refStart or refEnd would start or end the new bases within a deletion, this function will return null
      *
+     * If the cigar starts with an insertion, the inserted bases are considered as coming before the start position and
+     * are therefore excluded from the result.  That is getBasesCoveringRefInterval(0, 3, "ACTTGT", 0) should yield "TTGT".
```
[the remark this answers is not in the document]
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/utils/read/AlignmentUtils.java]: done
